### PR TITLE
add e2e test for hostNetwork pods with dnsPolicy: ClusterFirstWithHostNet

### LINKED ISSUE #112488: Add more e2e test cases for other DNSPolicies
https://github.com/kubernetes/kubernetes/blob/661899f1a14680840e06c84bc3ff46f73d203012/test/e2e/network/dns.go#L569

There are not e2e tests for some DNSPolicies

see: https://github.com/kubernetes/kubernetes/issues/112481

/help
/sig network

### PR DESCRIPTION
**Pull request by aojea**
/kind cleanup
```release-note
NONE
```

Ref: https://github.com/kubernetes/kubernetes/issues/112481
Fixes: https://github.com/kubernetes/kubernetes/issues/112488

### CODE REVIEW

**Comment by aojea** [comment]
/assign @MrHohn 
/hold

to check the test is running and passing, I don't know if it will picked by the presubmit, I think so, but I prefer to double check

File: test/e2e/network/dns.go [hunk 1]
```diff
@@ -569,4 +569,44 @@ var _ = common.SIGDescribe("DNS", func() {
...
+		jessieProbeCmd, jessieFileNames := createProbeCommand(namesToResolve, nil, regularService.Spec.ClusterIP, "jessie", f.Namespace.Name, framework.TestContext.ClusterDNSDomain, framework.TestContext.ClusterIsIPv6())
+		ginkgo.By("Running these commands on wheezy: " + wheezyProbeCmd + "\n")
+		ginkgo.By("Running these commands on jessie: " + jessieProbeCmd + "\n")
+
+		// Run a pod which probes DNS and exposes the results by HTTP.
+		ginkgo.By("creating a pod to probe DNS")
+		pod := createDNSPod(f.Namespace.Name, wheezyProbeCmd, jessieProbeCmd, dnsTestPodHostName, dnsTestServiceName)
+		pod.ObjectMeta.Labels = testServiceSelector
+		pod.Spec.HostNetwork = true
+		pod.Spec.DNSPolicy = v1.DNSClusterFirstWithHostNet
```
Comment (aojea) [inline test/e2e/network/dns.go:608]: this is the only different with `should resolve DNS of partial qualified names for services`

**Comment by aojea** [comment]
/test pull-kubernetes-unit

> k8s.io/kubernetes/pkg/controller/disruption: TestStalePodDisruption/stale_pod_disruption expand_more	0s
> k8s.io/kubernetes/pkg/controller/disruption: TestStalePodDisruption expand_more

/hold cancel

> Kubernetes e2e suite: [It] [sig-network] DNS HostNetwork should resolve DNS of partial qualified names for services on hostNetwork pods with dnsPolicy: ClusterFirstWithHostNet [LinuxOnly]

File: test/e2e/network/dns.go [hunk 2]
```diff
@@ -570,3 +570,49 @@ var _ = common.SIGDescribe("DNS", func() {
 	})
 
 })
+
+var _ = common.SIGDescribe("DNS HostNetwork", func() {
+	f := framework.NewDefaultFramework("hostnetworkdns")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
```
Comment (aojea) [inline test/e2e/network/dns.go:576]: TIL, it requires LevelPrivileged for hostNetwork pods

**Review by MrHohn — APPROVED** [review]
Thanks for adding this.
/lgtm

**Pull request merged**